modules/tinymce/src/plugins/anchor/main/ts/core/Anchor.ts

```diff
@@ -6,31 +6,33 @@
  */
 
 import Editor from 'tinymce/core/api/Editor';
+import { Element } from '@ephox/dom-globals';
 
-const isValidId = function (id: string) {
+const isAnchor = (editor: Editor, node: Element) => node.tagName === 'A' && editor.dom.getAttrib(node, 'href') === '';
+
+const isValidId = (id: string) => {
   // Follows HTML4 rules: https://www.w3.org/TR/html401/types.html#type-id
   return /^[A-Za-z][A-Za-z0-9\-:._]*$/.test(id);
 };
 
-const getId = function (editor: Editor) {
+const getId = (editor: Editor) => {
   const selectedNode = editor.selection.getNode();
-  const isAnchor = selectedNode.tagName === 'A' && editor.dom.getAttrib(selectedNode, 'href') === '';
-  return isAnchor ? (selectedNode.getAttribute('id') || selectedNode.getAttribute('name')) : '';
+  return isAnchor(editor, selectedNode) ? (selectedNode.getAttribute('id') || selectedNode.getAttribute('name')) : '';
 };
 
-const insert = function (editor: Editor, id: string) {
+const insert = (editor: Editor, id: string) => {
   const selectedNode = editor.selection.getNode();
-  const isAnchor = selectedNode.tagName === 'A' && editor.dom.getAttrib(selectedNode, 'href') === '';
 
-  if (isAnchor) {
+  if (isAnchor(editor, selectedNode)) {
     selectedNode.removeAttribute('name');
     selectedNode.id = id;
     editor.undoManager.add();
   } else {
     editor.focus();
     editor.selection.collapse(true);
-    editor.execCommand('mceInsertContent', false, editor.dom.createHTML('a', {
-      id
+    editor.insertContent(editor.dom.createHTML('a', {
+      id,
+      contenteditable: 'false'
```

Comment:
This is almost definitely incorrect. This shouldn't be contenteditable false, as that would break the `allow_html_in_named_anchor` setting for example.

Reply:
I'd be curious to know why you added it?

Reply:
I thought that it should be something that is always applied for generated anchors so added it explicitly here. Although, I didn't consider that it would negate the `allow_html_in_named_anchor` setting so will remove it.

Reply:
Tracked this down more. It's because this just generates the base HTML, so we don't want contenteditable false here since that should be controlled by other behaviour.

So what happens is `insertContent` will trigger the node filters, which then converts it to a cef element as required. That also brings up another point, as I noticed you removed the `!node.firstChild` check from `isAnchorNode`. That will definitely break the `allow_html_in_named_anchor` since it will then add the contenteditable false attribute, so that should be reinstated.

- [ ] Revert change to the `Anchor.isAnchorNode` check and maybe add some comments so others don't fall into this trap?

Reply:
Maybe add a comment why it's checking the first Child so it's not removed in the future.

Reply:
Another option, or it could be done in tandem is to also add a test, as this wasn't picked up as a failure so I'm assuming we don't have a test.